lib/plug_proxy/transport/cowboy.ex

```diff
@@ -123,7 +131,9 @@ defmodule PlugProxy.Transport.Cowboy do
   end
 
   defp before_send(%Plug.Conn{before_send: before_send} = conn, headers, state) do
-    conn = %{conn | resp_headers: headers, state: state}
-    Enum.reduce(before_send, conn, &(&1.(&2)))
+    conn = Enum.reduce(before_send, conn, & &1.(&2))
```

Comment:
the only change on the code is here
before_send callbacks must be executed before setting the state of the connection to :chunked,
otherwise Plug.Conn will raise